src/base/static/state/ducks/map.js

```diff
@@ -30,32 +30,67 @@ export const mapUpdatingFilterTargetLayerSelector = state => {
 
 // Actions
 const SET_LAYER_STATUS = "map/SET_LAYER_STATUS";
-const SET_BASEMAP_STATUS = "map/SET_BASEMAP_STATUS";
 const SET_MAP_POSITION = "map/SET_MAP_POSITION";
 const SET_MAP_SIZE_VALIDITY = "map/SET_MAP_SIZE_VALIDITY";
 const ACTIVATE_FEATURE_FILTER = "map/ACTIVATE_FEATURE_FILTER";
 const DEACTIVATE_FEATURE_FILTER = "map/DEACTIVATE_FEATURE_FILTER";
 const RESET_FEATURE_FILTER_GROUP = "map/RESET_FEATURE_FILTER_GROUP";
+const HIDE_OLD_BASEMAP = "map/HIDE_OLD_BASEMAP";
+const SHOW_BASEMAP = "map/SHOW_BASEMAP";
 
 // Action creators
-export const setLayerStatus = (layerId, layerStatus) => {
-  return {
+export const showLayers = (layerIds = []) =>
+  layerIds.map(layerId => ({
     type: SET_LAYER_STATUS,
     payload: {
-      layerId: layerId,
-      layerStatus: layerStatus,
+      id: layerId,
+      status: "loading",
+      isVisible: true,
     },
-  };
-};
-export const setBasemap = (layerId, layerStatus) => {
-  return {
-    type: SET_BASEMAP_STATUS,
+  }));
+export const hideLayers = (layerIds = []) =>
+  layerIds.map(layerId => ({
+    type: SET_LAYER_STATUS,
     payload: {
-      layerId: layerId,
-      layerStatus: layerStatus,
+      id: layerId,
+      isVisible: false,
     },
-  };
-};
+  }));
+export const initLayers = layers =>
+  layers.map(layer => ({
+    type: SET_LAYER_STATUS,
+    payload: {
+      id: layer.id,
+      isVisible: layer.type !== "place" ? !!layer.is_visible_default : false,
+      isBasemap: !!layer.is_basemap,
+      type: layer.type,
+      status: !!layer.is_visible_default ? "loading" : "",
+    },
+  }));
+export const setLayerLoaded = layerId => ({
+  type: SET_LAYER_STATUS,
+  payload: {
+    id: layerId,
+    isVisible: true,
+    status: "loaded",
+  },
+});
+export const setLayerError = layerId => ({
+  type: SET_LAYER_STATUS,
```

Comment:
I think we are unnecessarily overloading the `SET_LAYER_STATUS` action here. It seems like we should be using 3 action/reducer pairs instead of `SET_LAYER_STATUS`:
 * `UPDATE_LAYER_STATUS`
   * for the `setLayerError` and `setLayerLoaded` creators
   * reducer only sets the `status` attribute of the layer
 * `UPDATE_LAYER_VISIBLE`
   * for the `hideLayers` creator
   * reducer only sets the `isVisible` and attributes of the layer
 *  `LOAD_LAYER` 
   * for the `initLayers` creator
   * reducer sets the entire layer object

I think this is more consistent with the CRUD (or CRUL) pattern that Redux Ducks is establishing: https://github.com/erikras/ducks-modular-redux

This suggestion also assumes that we are going to fix the coupling between the `isVisible` and the `status` attributes. See my PR here on how we can address this: https://github.com/mapseed/platform/pull/1088/files

Also, I think it's super confusing to have a `layerStatus.status` attribute. Can we rename it to `layerStatus.loadStatus` or `layer.status`?

Lmk what you think - I may be missing something, but I'm always happy to discuss.

Reply:
Yes, I think this makes sense. I left a comment on #1088, but assuming that change is ok I will refactor the map duck to reflect the comments above.